include/deal.II/distributed/tria_base.h

```diff
@@ -132,6 +132,26 @@ namespace parallel
      */
     types::subdomain_id locally_owned_subdomain () const;
 
+    /**
+     * Returns a set of MPI ranks of the processors that have at least one
+     * ghost cell adjacent to the cells of the local processor. In other
+     * words, this is the set of subdomain_id() for all ghost cells.
+     *
+     * @note: If @p i is contained in the list of processor @p j, then @p j
+     * will also be contained in the list of processor @p i.
+     **/
+    virtual const std::set<unsigned int> &ghost_owners () const;
```

Comment:
If you don't need it now, remove the virtual marker. When I read code, if a function is virtual it always means to me that I will have to find other places where it may be overridden, and so it increases the complexity and interconnectedness of code. If it turns out later that you do need it to be virtual after all, you can always add the marker again.